text2emoji/models/test_model.py

```diff
@@ -23,29 +27,83 @@ def test_best_model(type):
 
+    device = "cuda" if torch.cuda.is_available() else "cpu"
+    
+    # Load best model
+    
+    best_model = torch.load("out/best_model.pt", map_location=torch.device('cpu'))
     best_model.to(device)
 
-    # Get accuracy
-    accuracy, loss = get_performance(
-        best_model, torch.tensor(test_embeddings).to(device), torch.tensor(test_labels).to(device)
+    # Get predictions of the model
+    probabilties = get_probabilities(
+        best_model, torch.tensor(eval_embeddings).to(device)
     )
-
-    print(f"Accuracy: {accuracy:.2f}")
-
-    # Find the most frequent label
-    unique_labels, counts = np.unique(test_labels, return_counts=True)
-    most_frequent_label = unique_labels[np.argmax(counts)]
-
-    # Find the accuracy of a model that always predicts the most frequent label
-    baseline_accuracy = (test_labels == most_frequent_label).mean()
-    print(f"Baseline accuracy: {baseline_accuracy}")
+    
+    # Get baseline accuracy (most frequent label)
+    most_freq_labels = np.full(len(eval_labels), np.argmax(np.bincount(eval_labels)))
+    most_freq_one_hot = np.zeros((len(eval_labels), len(np.unique(eval_labels))))
+    most_freq_one_hot[np.arange(len(eval_labels)), most_freq_labels] = 1
+    
+    # Get baseline accuracy (random, with same distribution as labels)
+    random_labels = np.random.choice(eval_labels, len(eval_labels))
+    random_one_hot = np.zeros((len(eval_labels), len(np.unique(eval_labels))))
+    random_one_hot[np.arange(len(eval_labels)), random_labels] = 1
+
+    results = pd.DataFrame(columns=["accuracy", "top_3_accuracy", "top_5_accuracy", "macro f1_score", "macro precision", "macro recall"])
+    results.loc["most_freq"] = [
+        np.mean(most_freq_labels == eval_labels),
+        top_k_accuracy_score(eval_labels, most_freq_one_hot, k=3),
+        top_k_accuracy_score(eval_labels, most_freq_one_hot, k=5),
+        f1_score(eval_labels, most_freq_labels, average="macro"),
+        precision_score(eval_labels, most_freq_labels, average="macro"),
+        recall_score(eval_labels, most_freq_labels, average="macro"),
+    ]
+    results.loc["random"] = [
+        np.mean(random_labels == eval_labels),
+        top_k_accuracy_score(eval_labels, random_one_hot, k=3),
+        top_k_accuracy_score(eval_labels, random_one_hot, k=5),
+        f1_score(eval_labels, random_labels, average="macro"),
+        precision_score(eval_labels, random_labels, average="macro"),
+        recall_score(eval_labels, random_labels, average="macro"),
+    ]
+    results.loc["model"] = [
+        np.mean(probabilties.argmax(axis=1) == eval_labels),
+        top_k_accuracy_score(eval_labels, probabilties, k=3),
+        top_k_accuracy_score(eval_labels, probabilties, k=5),
+        f1_score(eval_labels, probabilties.argmax(axis=1), average="macro"),
+        precision_score(eval_labels, probabilties.argmax(axis=1), average="macro"),
+        recall_score(eval_labels, probabilties.argmax(axis=1), average="macro"),
+    ]
```

Comment:
I think these statements can be simplified using a loop but it's not too big of a problem